### DISCUSSION TITLE
How to Set an Image as the Post Excerpt?

### DISCUSSION THREAD

**Discussion by orion-miller** [jekyll/jekyll#9128] [Q&A] [answered]
I'm trying to find a way to set an image as my post excerpt. The idea is that I can pick the best image from the post as the excerpt rather than just the first section of the post.

I have tried putting the image in the Excerpt property in the front matter, however it seems only text works for this field because it errors out, seemingly due to the "!" character.
`excerpt: ![Test](assets\images\2022-04-03-design-project-shock-dynomometer\571_574.png)`

I know there is also the excerpt separator property, but this only allows an endpoint to be defined for the excerpt but not the startpoint. I then tried to add this functionality using the method outlined here https://cjshelton.github.io/blog/2019/05/27/customising-jekyll-excerpt-start.html 
However this seems to append "-->" to each of my posts, and also shows duplicates of the image.

Does anyone know of a better way to pick a given image from the post as the excerpt? 
Thanks!

**Comment by pranabdas** [comment, marked as answer]
How about, in the front matter:
```
---
excerpt_img_url: "/assets/img/xxxx.png"
---
```
and where you want to render it:
```
![img alt]({{ page.excerpt_img_url }})
```

**Reply by orion-miller** [reply]
Came back around to this finally and got it to work. The page for my posts is built by a home.html file, so the only difference from the suggestion above was needing to use an html tag rather than Markdown to render the image. 

![image](https://user-images.githubusercontent.com/33074666/199868316-e4d9f884-c800-4ade-9481-cbe6f1d5e4e3.png)

Thanks for the suggestion!

**Comment by ashmaroli** [comment]
Technically, an image is not an excerpt of an article. But that could just be my opinionated view.
What theme are you using?
If you have the ability to edit your layouts, all you may need to do is swap instances of `{{ post.excerpt }}` with a corresponding Liquid construct to render an image like you have tried above.
